data_portal_api.yml

```diff
@@ -0,0 +1,334 @@
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+    delete:
+      summary: Delete a submissions
+      tags:
+        - submissions
+      description: >-
+        this call deletes the submission related to the submission UUID. This does not delete a project if it has been previously publicly published.
+      parameters:
+        - $ref: '#/components/parameters/submission_uuid'
+      responses:
+        '200':
+          $ref: '#/components/responses/200'
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+  /v1/submissions/{submission_uuid}/{filename}:
+    put:
+      summary: Add file to submission
+      tags:
+        - submissions
+      description: >-
+        this call adds a file to the project. If a file already exists, it will not replace it and instead return that a file already exists. A quick validation will be performed to ensure that if the file is a data file, then the extension is one of the accepted types (i.e. .loom, .h5ad, or Seurat).
+      parameters:
+        - $ref: '#/components/parameters/submission_uuid'
+        - $ref: '#/components/parameters/filename'
+        - $ref: '#/components/parameters/filetype'
+      responses:
+        '201':
+          $ref: '#/components/responses/201'
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '409':
+          $ref: '#/components/responses/409'
+    delete:
+      summary: Delete project file'
+      tags:
+        - submissions
+      description: >-
+        This call removes a file from the project if the file and project exists. If either don't exists, then a warning will be error will be returned.
+      parameters:
+        - $ref: '#/components/parameters/submission_uuid'
+        - $ref: '#/components/parameters/filename'
+      responses:
+        '200':
+          $ref: '#/components/responses/201'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+  /v1/submissions/validate/{project_uuid}:
+    post:
+      tags:
+        - submissions
+      summary: Validate the Project
+      description: >-
+        Validates the project according to the guidelines below in the Validation section and outputs the result, specifying all errors. The result of the validation is also stored in the backend database.
```

Comment:
Probably need to update this description so that it doesn't refer to the "Validation section" 😄